server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/features/ResetFeatureStateRequest.java

```diff
@@ -8,29 +8,77 @@
 
 package org.elasticsearch.action.admin.cluster.snapshots.features;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionRequest;
 import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.support.master.MasterNodeRequest;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
 
 /** Request for resetting feature state */
-public class ResetFeatureStateRequest extends ActionRequest {
+public class ResetFeatureStateRequest extends MasterNodeRequest<ResetFeatureStateRequest> {
+
+    private static final Version FEATURE_RESET_ON_MASTER = Version.V_8_0_0;
+
+    public static ResetFeatureStateRequest fromStream(StreamInput in) throws IOException {
```

Comment:
Dear reviewer, since this API is experimental, I am wondering if we should simply throw here and say "Feature reset is not currently available in this type of mixed cluster".

This way we don't worry about this sort of logic to translate between a regular ActionRequest and a MasterNodeRequest

Reply:
I'm good with this, we should have covered this case to start with and this API is definitely experimental. Plus, I believe, currently undocumented. I'd be very surprised to hear of anyone using the API in the wild right now.